PresnyaHedonist/Controllers/FeedVC.swift

```diff
@@ -0,0 +1,147 @@
+        navigationController?.navigationBar.titleTextAttributes = [NSAttributedString.Key.font: Fonts.vcHeads!]
+        title = VCTitles.feedTitle
+    }
+    
+    
+    func fetchData() {
+        do {
+            places = try context.fetch(Place.fetchRequest())
+            
+        } catch {
+            presentAlert(title: AlertTitle.error,
+                         message: Errors.fetchError)
+        }
+    }
+    
+    
+    func configureSearchController() {
+        searchController.searchResultsUpdater                   = self
+        searchController.searchBar.delegate                     = self
+        searchController.obscuresBackgroundDuringPresentation   = false
+        searchController.searchBar.placeholder                  = "Поиск..."
+        searchController.searchBar.scopeButtonTitles            = ["Все", "Ресторан", "Места", "Завтраки", "Бар"]
+        
+        navigationItem.searchController                         = searchController
+        definesPresentationContext                              = true
+    }
+    
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if tableViewFeed.indexPathForSelectedRow == nil {
+            return
+        }
+        
+        if let indexPath = tableViewFeed.indexPathForSelectedRow {
+            
+            var place: Place
+            
+            if isFiltering { place = filteredPlaces[indexPath.row] }
+            else { place = places[indexPath.row] }
+            
+            let destination = segue.destination as! DetailsVC
+            destination.place = place
+        }
+    }
+}
+
+
+extension FeedVC: UITableViewDelegate, UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if isFiltering { return filteredPlaces.count }
+        return places.count
+    }
+    
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: CellsID.FEED_CELL) as! FeedCell
+        
+        var place: Place
+        
+        if isFiltering { place = filteredPlaces[indexPath.row] }
+        else { place = places[indexPath.row] }
+        
+        cell.setCell(place)
+        return cell
+    }
+    
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableViewFeed.deselectRow(at: indexPath, animated: true)
+    }
+}
+
+
+extension FeedVC: UISearchResultsUpdating {
+    
+    func updateSearchResults(for searchController: UISearchController) {
+        let searchBar = searchController.searchBar
+        let scope = searchBar.scopeButtonTitles![searchBar.selectedScopeButtonIndex]
+        contentFilter(searchController.searchBar.text!, scope: scope)
+    }
+    
+    
+    func contentFilter(_ searchText: String, scope: String = "Все") {
+        
+        filteredPlaces = places.filter({ (place) -> Bool in
```

Comment:
Here you technically search/filter through the 'places' entities in memory. That is just fine in this case since the data set isn't huge, but since the data is in Core Data, one could also filter directly in the database.

Have you looked at NSFetchedResultsController? It can be hooked up to do fetching in batches and also to do searching (by adding an NSPredicate to the 'fetchRequest' of the controller).
Again, it doesn't matter with 80 or so instances, but perhaps it could be a good learning experience to do the same thing with a NSFetchedResultsController?

Reply:
Honestly, I'm not very acquainted with `NSFetchedResultsController`... But, it's the zone for growth! I definitely learn more and return to this point a bit later)